File: Programs/Test 1/task1.py
def solution(S):

## Alternate to remove '-' and ' '
##    num = ""
##    for index in range(len(S)):
##        if (S[index] != ' ') & (S[index] != '-'):
##            #print "S index =", S[index]
##            num = num + S[index]
##    print num

    phone = ""
    #print length
    str1 = S.replace(" ","")
    str2 = str1.replace("-","")
    print str2
    length = len(str2)
    print "Length: ", length

    if (length == 1): #phone length of 1
        phone = str2

    elif (length % 3 == 1): #remaining 1
        print "Remaining 1"
        for index in range(length-2):
            phone = phone + str2[index]
            #print "Index = ", index
            if ((index+1) % 3 == 0):
                phone = phone + '-'
        phone = phone + '-' + str2[length-2] + str2[length-1]

    else:
        print "Remaining 0 or 2"
        for index in range(length):
            phone = phone + str2[index]
            #print "Index = ", index
            if ((index+1) % 3 == 0) & (index != length-1):
                #print "Index = ", (index+1) % 3
                phone = phone + '-'

    print "Phone:", phone
    return

if __name__ == '__main__':
    S = "00-44  48 5555 861    223"
    S2 = "   0-1--- 055"
    S3 = "---1  "
    solution(S)
    print "-----------------------------------------"
    solution(S2)
    print "-----------------------------------------"
    solution(S3)

File: Programs/Test 1/task1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char * Solution(char *S, size_t length){
    printf("%s\n", S);
    char *str = (char*)malloc(length * sizeof(char)); //dynamic allocate size
    strcpy(str,S);
    printf("S= %s, Str= %s\n", S, str);
    char *phone = (char*)malloc(length * sizeof(char));
    char *str2 = (char*)malloc(length * sizeof(char));
    int len;
    int i;
    int count = 0;
    len = strlen(str);
    printf("Length: %d\n", len);
    for(i = 0; i < len; i++){ //remove dash and space
        if (str[i] != '-' && str[i] != ' '){
            //printf("%c", str[i]);
            str2[count] = str[i];
            count++;
        }
    }
    str2[count] = '\0'; //end string
    printf("\n");
    printf("Str2 %s\n", str2);
    int len2 = strlen(str2);
    printf("Len2 = %d\n", len2);
    //if length 1
    if (len2 == 1){
        strcpy(phone,str2);
    }
    //if remaining 1
    else if (len2 % 3 == 1){
        printf("Remaining 1 \n");
        count = 0;
        for (i = 0; i < (len2-2); i++){
            phone[count] = str2[i];
            count++;
            if ((i+1) % 3 == 0){
                phone[count] = '-';
                count++;
            }
        }
        phone[count] = '-';
        count++;
        phone[count] = str2[len2-2];
        count++;
        phone[count] = str2[len2-1];
        count++;
        phone[count] = '\0';
    }
    //if remaining 0 or 2
    else {
        printf("Remaining 0 or 2 \n");
        count = 0;
        for (i = 0; i < (len2); i++){
            phone[count] = str2[i];
            count++;
            if ((i+1) % 3 == 0 && (i+1) != len2){
                phone[count] = '-';
                count++;
            }
        }
        phone[count] = '\0';
    }
    printf("Phone: %s \n", phone);
    return phone;
}



int main(void){
    char S[] = "   1------23- -4 56 7-8 9 ";
    char S2[] = "   1------";
    char S3[] = " 1 ---- 2-3- 4  -5--67";
    char *ans = (char*)malloc(strlen(S) * sizeof(char));

    ans = Solution(&S, sizeof (S) / sizeof (*S)); //input size of array
    printf("ans= %s\n------------------------------ \n", ans);

    ans = Solution(&S2, sizeof (S2) / sizeof (*S2)); //input size of array
    printf("ans= %s\n------------------------------ \n", ans);

    ans = Solution(&S3, sizeof (S3) / sizeof (*S3)); //input size of array
    printf("ans= %s\n------------------------------ \n", ans);

    return 0;
}
